src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerExecutor.cs

```diff
@@ -234,6 +272,24 @@ public void ExecuteCompilationStartActions(ImmutableArray<CompilationStartAnalyz
             }
         }
 
+        private TAnalysisContext WithAnalyzerSpecificOptions<TAnalysisContext>(
+            TAnalysisContext context,
+            DiagnosticAnalyzer analyzer,
+            Func<TAnalysisContext, AnalyzerOptions, TAnalysisContext> withOptions)
+        {
+            // No specific options factory.  Can use the shared context.
+            if (_analyzerToCachedOptions is null)
+                return context;
```

Comment:
note: this is the common case for those not referencing hte roslyn sdk.  So there should be almost no overhead in that common case.